rs/nervous_system/agent/src/pocketic_impl.rs

```diff
@@ -5,6 +5,20 @@ use thiserror::Error;
 
 use crate::CallCanisters;
 
+/// A wrapper around PocketIc that specifies a sender for the requests.
+/// The name is an analogy for `ic_agent::Agent`, since each `ic_agent::Agent` specifies a sender.
+pub struct PocketIcAgent<'a> {
+    pub pocket_ic: &'a PocketIc,
+    pub sender: Principal,
```

Comment:
No change required.

One downside of this design direction is that you'll need a new agent for every principal.  Maybe that's already implied?

It kind of hides the ability of PocketIc and StateMachine to specify the sender in each request, which is very nice when you want to test canister-to-canister interaction or have a test that takes into account many principals.

I'm only mentioning this b/c it's a decision that will affect how the library gets used.  I don't know if that's really a concern or not because I haven't really used the library for much yet, so I can't say I have a strong opinion on whether or not this is generally consonant with the design and intention.  So do what seems right, I just wanted to point this out as a consideration.

Reply:
I agree with everything you've said. It's potentially something worth iterating on, but I think this is the way to go for now. Since `ic-agent` doesn't support customizing the sender on a per-request basis aside from passing in a different agent, we kind of have to imitate that I think.

Reply:
My 2 cents: As far as I know, the terminology is "agent = client + identity"

In that sense, an agent is always per principal.

Reply:
Sounds good, if that’s the pattern already then let’s just stick with it for this library